compose/init.sql

```diff
@@ -0,0 +1,446 @@
+              "regex": "^\\w+_\\w+_\\w+_.+$",
+              "weight": 3
+            },
+            {
+              "name": "Extension",
+              "description": "Filename must contain an extension, ex: phone_bill_robert_2024.pdf",
+              "regex": "\\.\\w+$",
+              "weight": 2.5
+            },
+            {
+              "name": "White Space",
+              "description": "Don''t accept white spaces",
+              "regex": "^\\S+$",
+              "weight": 2
+            },
+            {
+              "name": "Unauthorized Char",
+              "description": "Don''t accept special characters",
+              "regex": "^[A-Za-z0-9._]*$",
+              "weight": 2
+            }
+          ]
+        }'
+    );
+
+
+INSERT INTO rules (name, description, weight, rules_config)
+VALUES
+    ('perished',
+     'File is considered perished if it has not been modified for a certain period',
+     2.0,
+     '{
+       "type": "perished",
+       "expiration_days": 300
+     }'
+    );
+
+INSERT INTO rules (name, description, weight, rules_config)
+VALUES
+    ('duplicated',
+     'File is considered duplicated if it appears too many times in the system',
+     1.5,
+     '{
+       "type": "duplicated",
+       "max_occurrences": 3
+     }'
+    );
+
+
+-- TEST VALUES SHOULD BE DELETE
+INSERT INTO files (name, size, file_hash, last_modified, misnamed_score, perished_score, duplicated_score, global_score)
+VALUES
+    ('correct_mot1_mot2_2022.txt', 2025, 'ab', NOW(), 'U', 'U', 'U', 'U'),
+    ('only_2words_2023.txt', 2025, 'ab', '2024-02-05', 'U', 'U', 'U', 'U'),
+    ('nounderscore3mot1mot22024.csv', 2025, 'alone', '2023-10-05', 'U', 'U', 'U', 'U'),
+    ('onlyoneword.txt', 2025, 'abcd', '2023-12-05', 'U', 'U', 'U', 'U'),
+    ('nodate_mot1_mot2.txt', 2025, 'abcd', '2024-01-25', 'U', 'U', 'U', 'U'),
+    ('noextension_mot1_mot2_2010', 2025, 'abcd', '2024-05-05', 'U', 'U', 'U', 'U'),
+    ('4words_mot1_mot2_mot3_2013.txt', 2025, 'acd', '2023-02-05', 'U', 'U', 'U', 'U'),
+    ('correct_perished_mot2_2010.txt', 2025, 'acd', '2023-02-05', 'U', 'U', 'U', 'U'),
+    ('bad! ', 2025, 'abcd', '2023-02-05', 'U', 'U', 'U', 'U'),
+    ('special_aze_&ù\_2022.txt', 2025, 'abcd', '2023-02-05', 'U', 'U', 'U', 'U');
+-------------------------------------------------
+
+ALTER TABLE duplicate_associative_table
+    ADD CONSTRAINT unique_file_pair UNIQUE (original_file_id, duplicate_file_id);
+
+ALTER TABLE duplicate_associative_table
+    ADD CONSTRAINT no_self_duplicate CHECK (original_file_id != duplicate_file_id);
+
+CREATE OR REPLACE PROCEDURE detect_and_store_duplicates()
+    LANGUAGE plpgsql AS $$
+BEGIN
+    INSERT INTO duplicate_associative_table (original_file_id, duplicate_file_id)
+    SELECT f1.id, f2.id
+    FROM files f1
+             JOIN files f2 ON f1.file_hash = f2.file_hash AND f1.id < f2.id
+    ON CONFLICT DO NOTHING;
```

Comment:
What happens when a file is deleted a was the _original_ ? Same question applies when a new file is processed but is older than the others

Reply:
Nothing, because that code in proceed on certain file_id at instant T so if it has duplicates that store it, nothing otherwise

Reply:
Fair